cmake/kokkos_arch.cmake

```diff
@@ -845,15 +845,42 @@ IF (KOKKOS_ENABLE_OPENACC)
       Clang -Xopenmp-target=nvptx64-nvidia-cuda -march=${CLANG_CUDA_ARCH}
             -fopenmp-targets=nvptx64-nvidia-cuda
     )
+    IF(DEFINED ENV{CUDA_PATH})
+      COMPILER_SPECIFIC_LINK_OPTIONS(Clang -L$ENV{CUDA_PATH}/lib64)
+    ENDIF()
+    COMPILER_SPECIFIC_LIBS(
+      Clang -lcudart
+      NVHPC -cuda
+    )
   ELSEIF(KOKKOS_AMDGPU_ARCH_FLAG)
     COMPILER_SPECIFIC_FLAGS(
       Clang -Xopenmp-target=amdgcn-amd-amdhsa -march=${KOKKOS_AMDGPU_ARCH_FLAG}
             -fopenmp-targets=amdgcn-amd-amdhsa
     )
+    IF(DEFINED ENV{ROCM_PATH})
+      COMPILER_SPECIFIC_FLAGS(Clang -I$ENV{ROCM_PATH}/include)
+      COMPILER_SPECIFIC_LINK_OPTIONS(Clang -L$ENV{ROCM_PATH}/lib)
+    ENDIF()
+    COMPILER_SPECIFIC_LIBS(Clang -lamdhip64)
+  ELSEIF(KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE)
+    # Compile for kernel execution on the host. In that case,
+    # memory is shared between the OpenACC space and the host space.
+    COMPILER_SPECIFIC_FLAGS(
+      NVHPC -acc=multicore
+    )
+    COMPILER_SPECIFIC_DEFS(
+      NVHPC KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE
+    )
+    COMPILER_SPECIFIC_DEFS(
+      Clang KOKKOS_ENABLE_OPENACC_FORCE_HOST_AS_DEVICE
+    )
   ELSE()
+    # Automatic fallback mode; try to offload any available GPU, and fall back
+    # to the host CPU if no available GPU is found.
     COMPILER_SPECIFIC_FLAGS(
-      NVHPC -acc
+      NVHPC -acc=gpu,multicore
```

Comment:
Presumably this change is not actually needed per se.
Would you explain why it is preferable to specify the target architectures?
Is that a Clacc thing?
(I am not necessary objecting to the change, just wanting to make sure I understand)

Reply:
The behavior of `-acc` is not the same as that of `-acc=gpu,multicore`; when host fallback occurs, with `-acc`, OpenACC kernels are executed serially on the host CPU while executed in parallel with `-acc=gpu,multicore`. (GPU execution behavior is the same.)